whylog/constraints/verifier.py

```diff
@@ -0,0 +1,177 @@
+import itertools
+
+from whylog.config import Clue
+from whylog.constraints import IdenticalIntervals, TimeConstraint
+from whylog.constraints.exceptions import UnsupportedConstraintTypeError
+from whylog.front import FrontInput
+
+
+class ConstraintRegistration(object):
+    CONSTRAINTS = {
+        'identical': IdenticalIntervals({}),
+        'time': TimeConstraint({})
+        # register your constraint here
+    }
+
+    @classmethod
+    def get_constraint(cls, constraint_data):
+        if constraint_data['name'] in cls.CONSTRAINTS:
+            return cls.CONSTRAINTS[constraint_data['name']]
+        raise UnsupportedConstraintTypeError(constraint_data)
+
+
+class ConstraintManager(object):
+    """
+    there should be one such object per rule being verified
+    """
+
+    def __init__(self):
+        self._actual_constraints = {}
+
+    def __getitem__(self, constraint_data):
+        if not constraint_data['name'] in self._actual_constraints:
+            constraint_verifier = ConstraintRegistration.get_constraint(constraint_data)
+            self._actual_constraints[constraint_data['name']] = constraint_verifier
+        return self._actual_constraints[constraint_data['name']]
+
+
+class Verifier(object):
+    UNMATCHED = Clue(None, None, None, None)
+
+    @classmethod
+    def _front_input_from_clue(cls, clue):
+        return FrontInput(clue.line_offset, clue.line_prefix_content, clue.line_source)
```

Comment:
this should go to Clue, FrontInput or some factory. This should not be a Verifier-specific method.